src/Symfony/Component/HttpFoundation/ParameterBag.php

```diff
@@ -127,12 +127,34 @@ public function getDigits(string $key, string $default = ''): string
 
     /**
      * Returns the parameter value converted to integer.
+     * 
+     * @deprecated 6.1
      */
     public function getInt(string $key, int $default = 0): int
     {
+        trigger_deprecation('symfony/http-foundation', '6.1', 'Method "getInt" is deprecated because renamed to "convertInt".');
+
         return (int) $this->get($key, $default);
     }
 
+    /**
+     * Returns the parameter value converted to integer.
+     */
+    public function convertInt(string $key, int $default = 0): int
+    {
+        return $this->filter($key, $default, \FILTER_VALIDATE_INT, ['default' => $default, 'flags' => \FILTER_VALIDATE_INT]);
+    }
+
+    /**
+     * Returns the parameter value converted to string.
+     */
+    public function convertString(string $key, string $default = ''): string
+    {
+        $output = $this->get($key, $default);
+
+        return \is_array($output) ? $default : (string) $output;
```

Comment:
Looks like there's been some discussion around this already so apologies if this has been covered and I missed it.

Personally I'd find the behaviour of returning the default if the value is an array surprising. While it could be argued any usage *should* be covered by an applications tests, I'd prefer to see exceptions on any conditions not specified by the caller.

This would also apply to defaulting the value to `0` in `convertInt`, and `''` here, there doesn't seem to be a way to fail when an expected parameter is missing.